.github/workflows/cmd.yml

```diff
@@ -0,0 +1,112 @@
+            "https://api.github.com/repos/$ORG/$REPO/collaborators/$COLLABORATOR/permission")
+          
+          permission_level=$(echo "$permissions" | jq -r '.permission')
+          
+          if [[ "$permission_level" == "write" || "$permission_level" == "admin" ]]; then
+            echo "✅ User $COLLABORATOR has permission: "$permission_level". Proceeding..."
+          else
+            echo "❌ User $COLLABORATOR doesn't have enough permission: "$permission_level". Failing..." && exit 1
+          fi
+  acknowledge:
+    if: startsWith(github.event.comment.body, '/cmd')
+    needs: ["is-org-member"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Add reaction to triggered comment
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.reactions.createForIssueComment({
+              comment_id: ${{ github.event.comment.id }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              content: 'eyes'
+            })
+
+  update_typescript_api_augment:
+    needs: ["is-org-member"]
+    runs-on: self-hosted
+    env:
+      TMP_TARGET: "/tmp/target"
+      CARGO_TARGET_DIR: "target"
+      RUSTC_WRAPPER: "sccache"
+      CARGO_INCREMENTAL: "0"
+      SCCACHE_CACHE_SIZE: "100GB"
+      SCCACHE_GHA_ENABLED: "true"
+    # TODO: Uncomment before merge
+    #    if: ${{ github.event.comment.body == '/cmd generate-ts-api' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.set-tags.outputs.git_ref }}
+
+      - name: Cargo build
+        uses: ./.github/workflow-templates/cargo-build
+        with:
+          features: "fast-runtime"
+
+      - name: Pnpm
+        uses: pnpm/action-setup@v4.1.0
+        with:
+          version: 9
+
+      - name: "Setup Node"
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.x
+          cache: "pnpm"
+
+      # TODO: Extract to workflow template and re-use in other workflows (separate PR)
+      - name: "Generate Local interfaces"
+        run: |
+          chmod uog+x target/release/tanssi-node
+          chmod uog+x target/release/tanssi-relay
+          cd typescript-api
+          pnpm install --frozen-lockfile
+          pnpm create-local-interfaces
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Commit and push changes
+        run: |
+          if [[ -n "$(git status --porcelain)" ]]; then
+            cd typescript-api
+            git add .
+            echo "About to commit changes"
+            git commit -m "Auto-generated TypeScript interfaces"
+            echo "About to push changes"
+            git push -u origin $(git branch --show-current)
```

Comment:
Not sure if this branch name works, did you test it? Because not sure if the script ever checks out the branch, or only the commit. But if it checks out the branch, then just "git push" should also work?

Reply:
No, this part has not been tested yet, because there is no permissions to push. If we check the commit instead of branch, we can read the branch from the PR, will check.

Reply:
I tested this in a private repo and it pushed to master :joy: 

So for the first test, instead of git push let's use echo "Will push to $(git branch --show-current)" and not push